Not able to compile in release mode on 15.10

```
edjiang@ubuntu:~/throwaway-code/clibressl$ swift build -c release
Compile CLibreSSL strndup.c
/home/edjiang/throwaway-code/clibressl/Sources/CLibreSSL/strndup.c:26:1: error: expected identifier or '('
strndup(const char *str, size_t maxlen)
^
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:4: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
   ^
/home/edjiang/throwaway-code/clibressl/Sources/CLibreSSL/strndup.c:26:1: error: expected ')'
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:4: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
   ^
/home/edjiang/throwaway-code/clibressl/Sources/CLibreSSL/strndup.c:26:1: note: to match this '('
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:3: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
  ^
2 errors generated.
<unknown>:0: error: build had 1 command failures
error: exit(1): /home/edjiang/.swiftenv/versions/DEVELOPMENT-SNAPSHOT-2016-08-18-a/usr/bin/swift-build-tool -f /home/edjiang/throwaway-code/clibressl/.build/release.yaml
```

Works in debug mode. 

Works in 14.04

Tried in verbose mode on Ankit Aggarwal's recommendation: 

```
/usr/bin/clang -fPIC -fobjc-arc -fmodules -fmodule-name=CLibreSSL -fmodules-cache-path=/root/clibressl/.build/release/ModuleCache '-O2' -MD -MT dependencies -MF /root/clibressl/.build/release/CLibreSSL.build/strndup.c.d -c /root/clibressl/Sources/CLibreSSL/strndup.c -o /root/clibressl/.build/release/CLibreSSL.build/strndup.c.o -I /root/clibressl/Sources/CLibreSSL/include
/root/clibressl/Sources/CLibreSSL/strndup.c:26:1: error: expected identifier or '('
strndup(const char *str, size_t maxlen)
^
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:4: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
   ^
/root/clibressl/Sources/CLibreSSL/strndup.c:26:1: error: expected ')'
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:4: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
   ^
/root/clibressl/Sources/CLibreSSL/strndup.c:26:1: note: to match this '('
/usr/include/x86_64-linux-gnu/bits/string2.h:1319:26: note: expanded from macro 'strndup'
#   define strndup(s, n) __strndup (s, n)
                         ^
/usr/include/x86_64-linux-gnu/bits/string2.h:1300:3: note: expanded from macro '__strndup'
  (__extension__ (__builtin_constant_p (s) && __string2_1bptr_p (s)           \
  ^
2 errors generated.
```

--- a/Sources/CLibreSSL/strndup.c
+++ b/Sources/CLibreSSL/strndup.c
@@ -22,6 +22,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+#if !defined(__linux__)
+
 char *
 strndup(const char *str, size_t maxlen)
 {
@@ -37,3 +39,5 @@ strndup(const char *str, size_t maxlen)
 
 	return copy;
 }
+
+#endif

